### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by vikatyshchenko**
Created TAF structure. Added dependecies

### CODE REVIEW

**Review by sforktesting — APPROVED** [review]

File: src/main/java/com/report/portal/business/MyClass.java [hunk 1]
```diff
@@ -0,0 +1,8 @@
+package com.report.portal.business;
+
+public class MyClass {
+
+        public String concatenate(String one, String two){
```
[the remark this answers is not in the document]
Comment (vikatyshchenko) [reply src/main/java/com/report/portal/business/MyClass.java:5]: formatted!

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+  push:
+    branches: [ "main" ]
+
+
+jobs:
+  build:
+    name: Build
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
```
[the remark this answers is not in the document]
Comment (vikatyshchenko) [reply .github/workflows/build.yml:13]: tried. didn't work with v3

File: .github/workflows/gradle.yml [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+    - uses: actions/checkout@v3
+    - name: Set up JDK 17
+      uses: actions/setup-java@v3
+      with:
+        java-version: '17'
+        distribution: 'temurin'
+    - name: Run chmod to make gradlew executable
+      run: chmod +x ./gradlew
+    - name: Build with Gradle
+      uses: gradle/gradle-build-action@67421db6bd0bf253fb4bd25b31ebb98943c375e1
```
[the remark this answers is not in the document]
Comment (vikatyshchenko) [reply .github/workflows/gradle.yml:34]: tried. didn't work with v2

File: src/main/java/com/report/portal/core/exceptions/TestDataException.java [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+    public TestDataException(final String message) {
+        Log.error(message);
+        throw new RuntimeException(message);
+    }
+
+    public TestDataException(final String messageTemplate, Object... args) {
+        Log.error(messageTemplate, args);
+        throw new RuntimeException(String.format(messageTemplate, args));
+    }
+
```
[the remark this answers is not in the document]
Comment (vikatyshchenko) [reply src/main/java/com/report/portal/core/exceptions/TestDataException.java:16]: fixed!

File: .idea/gradle.xml [hunk 5]
```diff
@@ -1,10 +1,13 @@
 <?xml version="1.0" encoding="UTF-8"?>
```
[the remark this answers is not in the document]
Comment (vikatyshchenko) [reply .idea/gradle.xml]: removed!

File: .idea/uiDesigner.xml [hunk 6]
```diff
@@ -0,0 +1,124 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
[the remark this answers is not in the document]
Comment (vikatyshchenko) [reply .idea/uiDesigner.xml]: removed!

File: .github/workflows/build.yml (hunk 2 above)
[the remark this answers is not in the document]
Comment (dkatsechau) [reply .github/workflows/build.yml]: Actually it works well with @v3

File: .github/workflows/gradle.yml (hunk 3 above)
[the remark this answers is not in the document]
Comment (dkatsechau) [reply .github/workflows/gradle.yml]: Updated, works well with @v2

**Review by dkatsechau — APPROVED** [review]

**Pull request merged**